Compound_Risk_database.R

```diff
@@ -870,56 +865,111 @@ socio_forward <- inform_covid_warning %>%
+    ~ as.numeric(as.character(.))
   ) %>%
   mutate(
-    pov_prop_19_20 = ((y2020_prop / y2019_prop) -1) * 100,
-    pov_abs_19_20 = (y2020_prop - y2019_prop) * 100
-  ) %>%
+    pov_prop_22_21 = y2022 - y2021,
+    pov_prop_21_20 = y2021 - y2020,
+    pov_prop_20_19 = y2020 - y2019,
+    ) %>%
   filter(Label == "International poverty rate ($1.9 in 2011 PPP)") %>%
   rename_with(
     .fn = ~ paste0("S_", .),
     .cols = colnames(.)[!colnames(.) %in% c("Country")]
   ) 
 
-mpo_data <- normfuncpos(mpo_data, 50, 0, "S_pov_prop_19_20")
-mpo_data <- normfuncpos(mpo_data, 0.05, 0, "S_pov_abs_19_20")
+# Normalise based on percentiles
+mpo_data <- normfuncpos(mpo_data, 0.5, -1.5, "S_pov_prop_22_21")
+mpo_data <- normfuncpos(mpo_data, 1, -1, "S_pov_prop_21_20")
+mpo_data <- normfuncpos(mpo_data, 3, 0, "S_pov_prop_20_19")
+
+mpo_data <- mpo_data %>%
+  mutate(
+    S_pov_comb_norm = rowMaxs(as.matrix(dplyr::select(.,
+                                                      S_pov_prop_22_21_norm,
+                                                      S_pov_prop_21_20_norm, 
+                                                      S_pov_prop_20_19_norm))
+                              , na.rm = T)
+  )
+
+#-----------------------------—HOUSEHOLD HEATMAP FROM MACROFIN-------------------------------------
+# If Macro Fin Review is re-included above, we can reuse that. For clarity, moving data read here because it's not being used by macrosheet
+data <- read.csv("https://raw.githubusercontent.com/ljonestz/compoundriskdata/master/Indicator_dataset/macrofin.csv")
 
-#-----------------------------HOUSEHOLD HEATMAP FROM MACROFIN-------------------------------------
-household_risk <- macrosheet %>%
-  select(Country, M_Household.risks) %>%
+macrofin <- data %>%
+  mutate_at(
+    vars(Monetary.and.financial.conditions, contains("risk")),
+    funs(case_when(
+      . == "Low" ~ 0,
+      . == "Medium" ~ 0.5,
+      . == "High" ~ 1,
+      TRUE ~ NA_real_
+    ))) %>%
+  mutate(macrofin_risk = dplyr::select(., Spillover.risks.from.the.external.environment.outside.the.region:Household.risks) %>% rowSums(na.rm=T)) %>%
+  rename_with(
+    .fn = ~ paste0("M_", .),
+    .cols = colnames(.)[!colnames(.) %in% c("Country.Name","ISO3")]
+  ) %>%
+  rename(Country = ISO3) %>%
+  dplyr::select(-Country.Name)
+
+macrofin <- normfuncpos(macrofin, 2.1, 0, "M_macrofin_risk")
+
+household_risk <- macrofin %>%
+  dplyr::select(Country, M_Household.risks) %>%
   mutate(M_Household.risks = case_when(
     M_Household.risks == 0.5 ~ 7,
     M_Household.risks == 1 ~ 10,
     TRUE ~ M_Household.risks
   )) %>%
   rename(S_Household.risks = M_Household.risks)
 
-#----------------------------WB PHONE SURVEYS-----------------------------------------------------
-phone_data <- read_excel("~/Google Drive/PhD/R code/Compound Risk/formatted_data17 Nov 2020_internal.xlsx")
+#----------------------------—WB PHONE SURVEYS-----------------------------------------------------
```

Comment:
Excel structure and column names have changed. Code updated to match new version